scipy/ndimage/_multimethods.py

```diff
@@ -0,0 +1,1065 @@
+    """
+    def self_method(input, zoom, output=None, *args, **kwargs):
+        return (dispatchables[0], zoom, dispatchables[1]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_zoom_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def zoom(input, zoom, output=None, order=3, mode='constant', cval=0.0,
+         prefilter=True, *, grid_mode=False):
+    return input, _mark_non_coercible(output)
+
+
+def _rotate_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for ``rotate``.
+    """
+    def self_method(input, angle, axes=(1, 0), reshape=True,
+                    output=None, *args, **kwargs):
+        return (dispatchables[0], angle, axes, reshape,
+                dispatchables[1]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_rotate_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def rotate(input, angle, axes=(1, 0), reshape=True, output=None, order=3,
+           mode='constant', cval=0.0, prefilter=True):
+    return input, _mark_non_coercible(output)
+
+
+############################################
+# measurement multimethods
+############################################
+
+
+def _input_structure_output_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving required
+    ``input`` with ``structure`` and ``output`` kwargs.
+    """
+    def self_method(input, structure=None, output=None, *args, **kwargs):
+        return (dispatchables[0], dispatchables[1],
+                dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_structure_output_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def label(input, structure=None, output=None):
+    return input, structure, _mark_non_coercible(output)
+
+
+def _input_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving
+    the required ``input`` as the only dispatchable.
+    """
+    def self_method(input, *args, **kwargs):
+        return (dispatchables[0],) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def find_objects(input, max_label=0):
+    return (input,)
+
+
+def _input_labels_index_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for signatures involving required
+    ``input`` with ``labels`` and ``index`` kwargs.
+    """
+    def self_method(input, labels=None, index=None, *args, **kwargs):
+        return (dispatchables[0], dispatchables[1],
+                dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def labeled_comprehension(input, labels, index, func, out_dtype,
+                          default, pass_positions=False):
+    return input, labels, index
```

Comment:
```suggestion


def _labeled_comprehension_replacer(args, kwargs, dispatchables):
    """
    uarray argument replacer for signatures involving required
    ``input`` with ``labels`` and ``index`` kwargs.
    """
    def self_method(input, labels, index, func, out_dtype, *args,
                    **kwargs):
        return (dispatchables[0], dispatchables[1], dispatchables[2], func,
                dispatchables[3]) + args, kwargs

    return self_method(*args, **kwargs)


@_create_ndimage(_labeled_comprehension_replacer)
@all_of_type(np.ndarray)
@_get_docs
def labeled_comprehension(input, labels, index, func, out_dtype,
                          default, pass_positions=False):
    return input, labels, index, Dispatchable(out_dtype, np.dtype)
```

I think this should also dispatch on `out_dtype` as above?

Reply:
Fixed now.